File: src/exam-core/src/main/java/com/wcpdoc/exam/core/entity/MyCourseMaterial.java
package com.wcpdoc.exam.core.entity;

import java.time.LocalTime;
import java.util.Date;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;

import lombok.Data;

/**
 * 我的课程资料实体
 * 
 * v1.0 zhanghc 2026-01-12 10:03:53
 */
@Data
@TableName(value = "EXM_MY_COURSE_MATERIAL", autoResultMap = true)
public class MyCourseMaterial {
	@TableId(type = IdType.AUTO)
	private Integer id;
	private Integer courseId;
	private Integer courseMaterialId;
	private Integer userId;
	private String name;
	private String content;
	private Integer videoFileId;
	private LocalTime videoTime;
	private Integer questionNum;
	private Integer no;
	private LocalTime watchTime;
	private LocalTime activeTime;
	private Integer state;
	private Integer updateUserId;
	private Date updateTime;
}

File: m/ts/my-course-material.ts
import type { MyCourseQuestion } from './my-course-question';

// 我的课程资料接口
export interface MyCourseMaterial {
	courseId: number | null; // 课程ID
	courseMaterialId: number | null; // 课程资料ID
	userId: number | null; // 用户ID
	name: string; // 名称
	content: string; // 简介
	videoFileId: number | null; // 视频附件ID
	videoTime: string; // 视频时长
	no: number | null; // 第几小节
	watchTime: string; // 观看时间
	state: number | null; // 状态（0：删除；1：完成；2：未开始；3：进行中）
	questions: MyCourseQuestion[];
}
